tedge/src/cli/mqtt/publish.rs

```diff
@@ -0,0 +1,75 @@
+use crate::cli::mqtt::MqttError;
+use crate::command::{Command, ExecutionContext};
+use futures::future::FutureExt;
+use mqtt_client::{Client, Message, QoS, Topic};
+use std::time::Duration;
+use tokio::select;
+
+pub struct MqttPublishCommand {
+    pub topic: String,
```

Comment:
change type of `topic` to `mqtt_client::Topic` and do the "parsing" of the topic in the BuildCommand.